### PR TITLE
errata-webhook: add additional unit tests, refactor, and a fix

### PR DESCRIPTION
**Pull request by Davoska**
Additional unit tests for the errata-webhook.

- New unit tests mainly focused on the `public_errata_uri` and the `process_message` functions.
- Refactoring of the initial unit tests.
- Refactoring the `errata.py` to be more testable.
- Fixing a potential bug in the `errata.py`.

Pull request related to https://issues.redhat.com/browse/OTA-356.

### CODE REVIEW

File: hack/errata.py [hunk 1]
```diff
@@ -55,45 +55,56 @@ def run(poll_period=datetime.timedelta(seconds=3600),
         githubtoken=None,
         **kwargs):
     next_time = datetime.datetime.now()
+    if excluded_cache is None:
```
Comment (vrutkovs) [inline hack/errata.py:58]: I guess it would be easier to set `excluded_cache={}` in function declaration instead.

**Review by vrutkovs — APPROVED** [review]

**Comment by vrutkovs** [comment]
/hold

For Trevor's review

File: hack/errata.py [hunk 2]
```diff
@@ -55,45 +55,56 @@ def run(poll_period=datetime.timedelta(seconds=3600),
...
         next_time += poll_period
         _LOGGER.debug('sleep until {}'.format(next_time))
         time.sleep((next_time - datetime.datetime.now()).seconds)
 
 
+def process_message(message, cache, excluded_cache, webhook, githubrepo, githubtoken):
+    synopsis_match = _SYNOPSIS_REGEXP.match(message['synopsis'])
+    if not synopsis_match:
+        if message['synopsis'] not in excluded_cache:
+            _LOGGER.debug('{fulladvisory} shipped {when} does not match synopsis regular expression: {synopsis}'.format(**message))
```
Comment (wking) [inline hack/errata.py:80]: might make testing and logging easier to raise `ValueError` exceptions for these:

```python
def process_message(message, cache, excluded_cache, webhook, githubrepo, githubtoken):
    if excluded_cache and message['synopsis'] in excluded_cache:
        return
    synopsis_match = _SYNOPSIS_REGEXP.match(message['synopsis'])
    if not synopsis_match:
        raise ValueError('{fulladvisory} shipped {when} does not match synopsis regular expression: {synopsis}'.format(**message))
    ...
```

with:

```python
for message in poll(period=2*poll_period, **kwargs):
    try:
        process_message(...)
    except ValueError as error:
        _LOGGER.warn(error)
```

**Review by vrutkovs — APPROVED** [review]
/lgtm

Lets have Trevor have another look and pull the hold

**Comment by vrutkovs** [comment]
No objections :)

/hold cancel

**Pull request merged**